### PR TITLE
feat: [TKC-1055] test suite steps

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 

Adding execution requests to test suite steps

Note: depends on testkube-operator changes

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — COMMENTED** [review]
can we remove compilation errros first?

**Comment by vLia** [comment]
@vsukhin updated the code with the `testkube-operator` version containing the changes this PR depends on. However, the linter cache is acting up again

File: pkg/api/v1/testkube/model_test_step_execution_request.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * API version: 1.0.0
+ * Contact: testkube@kubeshop.io
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package testkube
+
+// test step execution request body
+type TestStepExecutionRequest struct {
+	// test execution custom name
+	Name string `json:"name,omitempty"`
```
Comment (vsukhin) [inline pkg/api/v1/testkube/model_test_step_execution_request.go:15]: should be regenerated

**Review by vsukhin — APPROVED** [review]
will we add support to cli command?

File: pkg/mapper/testsuites/kube_openapi.go [hunk 2]
```diff
@@ -80,7 +81,8 @@ func mapCRStepToAPI(crstep testsuitesv3.TestSuiteStepSpec) (teststep testkube.Te
 	switch true {
 	case crstep.Test != "":
 		teststep = testkube.TestSuiteStep{
-			Test: crstep.Test,
+			Test:             crstep.Test,
+			ExecutionRequest: testsuitestcl.MapTestStepExecutionRequestCRDToAPI(crstep.ExecutionRequest),
```
Comment (vsukhin) [inline pkg/mapper/testsuites/kube_openapi.go:85]: I prefer to add comments, that below is Pro code

File: pkg/api/v1/testkube/model_test_step_execution_request.go (hunk 1 above)
Comment (vLia) [reply pkg/api/v1/testkube/model_test_step_execution_request.go]: really good catch, the type was renamed and the generator did not delete the file

File: docs/docs/testkube-pro/articles/running-parallel-tests-with-test-suite.md [hunk 3]
```diff
@@ -26,4 +26,83 @@ For this test suite, we have added 5 tests that all run in parallel:
...
+![Test and Order of Execution](../../img/test-and-order-of-execution.png)
+
+## Test Suite Steps
+
+Test Suite Steps can be of two types:
+
+1. Tests: tests to be run.
+2. Delays: time delays to wait in between tests.
+
+Similarly to running a Test, running a Test Suite Step based on a test allows for specific execution request parameters to be overwritten. Step level parameters overwrite Test Suite level parameters, which in turn overwrite Test level parameters. The Step level parameters are configurable only via CRDs at the moment.
```
Comment (jfermi) [inline docs/docs/testkube-pro/articles/running-parallel-tests-with-test-suite.md:38]: Similar to running a Test,

File: docs/docs/testkube-pro/articles/running-parallel-tests-with-test-suite.md [hunk 4]
```diff
@@ -26,4 +26,83 @@ For this test suite, we have added 5 tests that all run in parallel:
...
+| pvcTemplateReference               | ✓    | ✓          | ✓         |
+| envConfigMaps                      | ✓    |            |           |
+| envSecrets                         | ✓    |            |           |
+| runningContext                     | ✓    | ✓          | ✓         |
+| slavePodRequest                    | ✓    |            |           |
+| secretUUID                         |      | ✓          |           |
+| labels                             |      | ✓          |           |
+| timeout                            |      | ✓          |           |
+
+Similarly to Tests and Test Suites, Test Suite Steps can also have a field of type `executionRequest` like in the example below:
```
Comment (jfermi) [inline docs/docs/testkube-pro/articles/running-parallel-tests-with-test-suite.md:84]: Similar to Tests and Test Suites,

**Review by jfermi — COMMENTED** [review]
A couple of editing suggestions.

**Pull request merged**